lectures/c++/03_more_on_pointers_and_vectors/readme.md

```diff
@@ -0,0 +1,99 @@
+# Lecture 03_more_on_pointers_and_vectors
+
+
+
+
+## 01_auto.cc
+
+[link to file](./01_auto.cc)
+
+This program introduces the `auto` specifier. When you declare a variable with this keyword
+its type will be automatically deduced by the compiler.
+
+PPP: chap 20.5.2
+CPL: chap 2.2.2, chap 6.3.6
+
+
+
+## 02_const_and_pointers.cc
+
+[link to file](./02_const_and_pointers.cc)
+
+This example teaches you how to declare a `const` pointer or a pointer to
+a `const` object. It also introduces the usage of the *sentinel* method to handle
+arrays in c++.
+
+CPL: chap 7.5
+
+
+
+## 03_special_pointers.cc
+
+[link to file](./03_special_pointers.cc)
+
+Here you can find more about pointers, like the `nullptr` keyword, the `static_cast`
+function, that can be used to safely perform conversions between compatible types,
+and the pointers to functions. `decltype()` is a function that returns the type of its argument.
+
+PPP: chap 17.4.5 nullptr, appendix A.5.7 static_cast, chap 27.2.5 pointer to func
+CPL: chap 2.2.5 nullptr, chap 6.3.6.3 decltype, chap 11.5.2 static_cast, chap 12.5 pointer to func
+
+
+
+
+## 04_command_line_args.cc
+
+[link to file](./04_command_line_args.cc)
+
+This is a small example on how you can pass arguments to a program from the command
+line.
+
+CPL: chap 10.2.7
+
+
+
+
+## 05_matrices.cc
+
+[link to file](./05_matrices.cc)
+
+This example shows how to deal with a matrix object defined as `int mat[][]` and how to pass
```

Comment:
Scriverei solo qualcosa del tipo

This example shows the basics of how to implement the mathematical concept of matrix.